src/test/kotlin/carracing/CarRaceTest.kt

```diff
@@ -0,0 +1,37 @@
+package carracing
+
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertNotEquals
+
+class CarRaceTest {
+    @Test
+    fun `when number of cars are less than 2 should throw exception with appropriate message`() {
+        val race = CarRace(numCars = 1, numRounds = 3)
+        val exception = assertFailsWith<IllegalArgumentException> {
+            race.startRace()
+        }
+        assertEquals("Not enough cars to race", exception.message)
+    }
+
+    @Test
+    fun `when number of rounds is zero should throw exception with appropriate message`() {
+        val race = CarRace(numCars = 3, numRounds = 0)
+        val exception = assertFailsWith<IllegalArgumentException> {
+            race.startRace()
+        }
+        assertEquals("Select at least one round to start the race", exception.message)
+    }
+
+    @Test
+    fun `race starts successfully when conditions are met`() {
```

Comment:
This test has potential to fail, since car movement depends on a random number.

Just like you extracted the random number out of `Car` to make `move()` easier to test, how about passing in the movement decision instead?
That way, `Car` only needs to know whether it should move, and doesn't need to know why.